src/link_prober/IcmpPayload.cpp

```diff
@@ -38,6 +38,28 @@ boost::uuids::uuid IcmpPayload::mGuid;
 uint32_t IcmpPayload::mCookie = 0x47656d69;
 uint32_t IcmpPayload::mVersion = 0;
 
+//
+// ---> TlvCommand();
+//
+// struct TlvCommand constructor
+//
+TlvCommand::TlvCommand() :
+    type(tlvtype), length(htons(1)), command(static_cast<uint8_t> (Command::COMMAND_SWITCH_ACTIVE))
```

Comment:
needs to be packed.

I suggest adding a generic class with set_type(), set(length), set_value(), and the value field could be an union of {uint8_t value[]; command_t cmd; ...}.

Reply:
For the tlv with variable size, will it hard for `union` to describe its contents?

Reply:
updated, please check